app/controllers/microservices_engine/v1/data_controller.rb

```diff
@@ -34,18 +34,25 @@ def update
         data = params['content']
         if data.present?
           data.each do |endpoint|
-            existing = Connection.where(object: endpoint['object'])
+            existing = Connection.where(object: endpoint['object']).first
             if existing.present?
               if endpoint['url'].present?
                 # URL exists so we will update as usual
-                existing.update_attributes(name: endpoint['name'], url: endpoint['url'])
+                existing.update_attributes(
+                  name: endpoint.require(:name),
+                  url: endpoint.require(:url)
+                )
               else
                 # URL is blank, thus the endpoint no longer exists
                 # Thus, we will remove the object as to avoid confusion
-                Connection.destroy(existing)
+                Connection.destroy(existing.id)
               end
-            else
-              new_connection = Connection.create(data)
+            elsif endpoint['url'].present?
+              new_connection = Connection.create(
+                name: endpoint.require(:name),
```

Comment:
Well, I figure the endpoints may want a label outside of their `url`. There's no 'significance' outside of pure convenience.